doc/logging/persistent_logger/persistent_loglevel.md

```diff
@@ -0,0 +1,217 @@
+
+
+# List of Figures
+* [persistent logger flow](#321-return-to-default-log-level)
+
+
+# Revision
+| Rev | Date     | Author          | Change Description                 |
+|:---:|:--------:|:---------------:|------------------------------------|
+| 0.1 | 06/21/22 | Eden Grisaro    | Initial version                    |
+
+# Scope
+This document provides an overview of the implementation of making the SWSS Logger persistent for SWSS, Syncd, and SAI components.
+
+# Motivation
+Log level verbosity is part of the configuration of the OS. Today, the log level is not persistent and gets a default value after reboot. It is required to add the ability to make the loglevel persistent.
+
+# Definitions/Abbreviation
+| Abbreviation  | Description                               |
+|---------------|-------------------------------------------|
+| SONiC         | Software for open networking in the cloud | 
+| SAI           | Switch Abstraction Interface              |
+
+
+# 1 Background
+Today, the user can configure the log level verbosity to each component in SWSS, Syncd, and SAI. In order to configure the log level in runtime, the user uses the **"swssloglevel" script**. The script updates the **LOGLEVEL DB** with the new verbosity. After reboot, the LOGLEVEL DB flushes, and the log level value of all components returns to default.
+
+## 1.1 "swssloglevel" usage
+
+The "swssloglevel" options:
+
+```
+admin@sonic:~$ swssloglevel -h
+Usage: swssloglevel [OPTIONS]
+SONiC logging severity level setting.
+
+Options:
+	 -h	print this message
+	 -l	loglevel value
+	 -c	component name in DB for which loglevel is applied (provided with -l)
+	 -a	apply loglevel to all components (provided with -l)
+	 -s	apply loglevel for SAI api component (equivalent to adding prefix "SAI_API_" to component)
+	 -p	print components registered in DB for which setting can be applied
+
+Examples:
+	swssloglevel -l NOTICE -c orchagent # set orchagent severity level to NOTICE
+	swssloglevel -l SAI_LOG_LEVEL_ERROR -s -c SWITCH # set SAI_API_SWITCH severity to ERROR
+	swssloglevel -l SAI_LOG_LEVEL_DEBUG -s -a # set all SAI_API_* severity to DEBUG
+
+```
+
+## 1.2 LOGLEVEL DB schema
+
+The LOGLEVEL DB is database #3. It contains a table for each component.
+Here is an example of the orchagent component table:
+
+```
+admin@sonic:~$ redis-cli -n 3 HGETALL "orchagent:orchagent"
+
+1) “LOGLEVEL”
+
+2) “INFO”
+
+3) “LOGOUTPUT”
+
+4) “SYSLOG”
+```
+
+
+# 2 Requirements Overview
+
+The persistent Logger should meet the following high-level functional requirement:
+- The user will be able to save the configuration of the loglevel and make it persistent after reboot. The default will be no persistent.
+
+# 3 Persistent log level design
+## 3.1 High-level design
+
+In the begining we consider two diffrent design aproaches for making the loglevel persistent to reboot:
+1. Keep using the existing LOGLEVEL DB and make it persistent by adding loglevel_db.json file. 
```

Comment:
1. Keep using the existing LOGLEVEL DB and make it persistent by adding loglevel_db.json file that will keep LOGLEVEL DB content

Reply:
Done